pkg/virt-api/webhooks/validating-webhook/admitters/vmi-update-admitter.go

```diff
@@ -232,28 +232,43 @@ func verifyHotplugVolumes(newHotplugVolumeMap, oldHotplugVolumeMap map[string]v1
 					})
 				}
 				disk := newDisks[k]
-				if disk.Disk == nil && disk.LUN == nil {
+				var bus v1.DiskBus
+
+				switch {
+				case disk.Disk != nil:
+					bus = disk.Disk.Bus
+					if bus != v1.DiskBusVirtio && bus != v1.DiskBusSCSI {
+						return webhookutils.ToAdmissionResponse([]metav1.StatusCause{
+							{
+								Type:    metav1.CauseTypeFieldValueInvalid,
+								Message: fmt.Sprintf("hotplugged Disk %s does not use a scsi or virtio bus", k),
+							},
+						})
+					}
+				case disk.LUN != nil:
+					bus = disk.LUN.Bus
+					if bus != v1.DiskBusSCSI {
+						return webhookutils.ToAdmissionResponse([]metav1.StatusCause{
+							{
+								Type:    metav1.CauseTypeFieldValueInvalid,
+								Message: fmt.Sprintf("hotplugged Disk %s does not use a scsi bus", k),
+							},
+						})
+					}
+				default:
 					return webhookutils.ToAdmissionResponse([]metav1.StatusCause{
 						{
 							Type:    metav1.CauseTypeFieldValueInvalid,
 							Message: fmt.Sprintf("Disk %s requires diskDevice of type 'disk' or 'lun' to be hotplugged.", k),
 						},
 					})
 				}
-				if (disk.Disk == nil || disk.Disk.Bus != "scsi") && (disk.LUN == nil || disk.LUN.Bus != "scsi") {
-					return webhookutils.ToAdmissionResponse([]metav1.StatusCause{
-						{
-							Type:    metav1.CauseTypeFieldValueInvalid,
-							Message: fmt.Sprintf("hotplugged Disk %s does not use a scsi bus", k),
-						},
-					})
 
-				}
-				if disk.DedicatedIOThread != nil && *disk.DedicatedIOThread {
+				if disk.DedicatedIOThread != nil && *disk.DedicatedIOThread && bus != v1.DiskBusVirtio {
 					return webhookutils.ToAdmissionResponse([]metav1.StatusCause{
 						{
 							Type:    metav1.CauseTypeFieldValueInvalid,
-							Message: fmt.Sprintf("hotplugged Disk %s can't use dedicated IOThread: scsi bus is unsupported.", k),
+							Message: fmt.Sprintf("hotplugged Disk %s can't use dedicated IOThread: only virtio bus is unsupported.", k),
```

Comment:
not sure this msg is so clear.. and I suppose it should have been "supported" at the end?
maybe:
"hotplugged Disk %s with dedicated IOThread requires virtio bus"
or
"hotplugged Disk %s can't use dedicated IOThread with bus different than virtio"
?